google/cloud/spanner/internal/spanner_stub.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_INTERNAL_SPANNER_STUB_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_SPANNER_INTERNAL_SPANNER_STUB_H_
+
+#include "google/cloud/spanner/client_options.h"
+#include "google/cloud/spanner/version.h"
+#include "google/cloud/status.h"
+#include "google/cloud/status_or.h"
+#include <google/spanner/v1/spanner.pb.h>
+#include <grpcpp/grpcpp.h>
+#include <memory>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+
+/**
+ * SpannerStub is a thin stub layer that avoids exposing the underlying
+ * transport stub directly (similar to what GAPIC will generate).
+ */
+class SpannerStub {
```

Comment:
For our future selves and other maintainers, it may be convenient to drop a link to the spanner.proto file in this comment. Here it is https://github.com/googleapis/googleapis/blob/master/google/spanner/v1/spanner.proto